swap/src/settings.rs

```diff
@@ -0,0 +1,177 @@
+use crate::{bitcoin::Timelock, config::File};
+use conquer_once::Lazy;
+use std::time::Duration;
+use url::Url;
+
+pub struct Settings {
+    pub wallets: Wallets,
+    pub protocol: Protocol,
+}
```

Comment:
I don't think that the wallet configuration and protocol settings should be bundled together.

The wallets settings are only used to instantiate the wallets where the protocol ones are carried around.

Reply:
I did that to ensure that both are initialized for `testnet`. If we separate it, we could run into the issue of mixing networks. But that is probably not enough to justify the `Settings` struct. I can remove it once we merged the changes on top into this PR.